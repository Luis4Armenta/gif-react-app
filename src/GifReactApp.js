import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifReactApp = ({ defaultCategories = [] }) => {
  const [categories, setCategories] = useState( defaultCategories );

  return (
    <Fragment>
      <h2>GifReactApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr />

      <ol>
        {
          categories.map(category => (
            <GifGrid 
              key={ category }
              category={ category } 
            />
          ))
        }
      </ol>
    </Fragment>
  )
}
