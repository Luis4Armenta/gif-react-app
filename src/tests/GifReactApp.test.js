import { shallow } from "enzyme"
import { GifReactApp } from '../GifReactApp'

describe('Probando <GifReactApp /> component', () => {
  test('Debe de mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifReactApp />);
  
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista de gategorias', () => {
    const categories = ['cats', 'cookies'];
    const wrapper = shallow(<GifReactApp defaultCategories= { categories }/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
  
  
})
