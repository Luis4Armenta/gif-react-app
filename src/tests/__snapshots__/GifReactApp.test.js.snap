// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifReactApp /> component Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifReactApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Probando <GifReactApp /> component debe de mostrar una lista de gategorias 1`] = `
<Fragment>
  <h2>
    GifReactApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="cats"
      key="cats"
    />
    <GifGrid
      category="cookies"
      key="cookies"
    />
  </ol>
</Fragment>
`;
